src/thor/bidirectional_astar.cc

```diff
@@ -123,21 +123,28 @@ bool BidirectionalAStar::ExpandForward(GraphReader& graphreader,
     return false;
   }
   const NodeInfo* nodeinfo = tile->node(node);
-  if (!costing_->Allowed(nodeinfo)) {
-    return false;
-  }
+
+  // Keep track of superseded edges
+  uint32_t shortcuts = 0;
 
   // Update the time information even if time is invariant to account for timezones
   auto seconds_offset = invariant ? 0.f : pred.cost().secs;
   auto offset_time = from_transition
                          ? time_info
                          : time_info.forward(seconds_offset, static_cast<int>(nodeinfo->timezone()));
 
-  uint32_t shortcuts = 0;
-  EdgeMetadata meta = EdgeMetadata::make(node, nodeinfo, tile, edgestatus_forward_);
+  // If we encounter a node with an access restriction like a barrier we allow a uturn
+  if (!costing_->Allowed(nodeinfo)) {
```

Comment:
This change handles the case where we encounter a disallowed node and instead of bailing we now want to consider making a uturn.

Reply:
the fallout from this is that we will enqueue more edges but practically speaking it doesnt really affect the number of iterations we run. so basically we'll have more labels but do pretty much the same amount of expansion